x-pack/plugins/observability_solution/apm/public/components/routing/templates/apm_main_template.tsx

```diff
@@ -118,21 +119,52 @@ export function ApmMainTemplate({
     isServerless: config?.serverlessOnboarding,
   });
 
+  let screenDescription = '';
```

Comment:
Could we move the AI assistant code to its own file?

Reply:
Same comments for the other files. I think we could clean up the components.

Reply:
Not really. This functionality is intended for other plugins to give contextual information to the assistant about what the user is looking at right now.

Reply:
I think what @cauemarcondes means is that it could be a separate hook or something similar. I've also done that in other places in APM (e.g. `getThroughputScreenContext`).

Reply:
IMO jumping between files hinders readability and leads engineers to just leave it alone because they don't see the contents anymore (many, many examples of this in the Kibana codebase). 

As stated though I will update.

Reply:
>IMO jumping between files hinders readability and leads engineers to just leave it alone because they don't see the contents anymore

I tend to agree and disagree with that 😄. The screen context/description is very specific for the AI assistant to know about this page/component. Me as a developer who is actively working on these components/pages I don't care what the assistant is doing. I know they are there though. What I want is to see a clean component, not having a bunch of static text on my way 😆.

For you, I think it'll be even easier as you'll have a clean file with only AI stuff.